### ISSUE TITLE
GitHub intercepting LFS uploads, refusing to actually upload despite pointer file.

### ISSUE THREAD

**Issue by kirkpatt** [git-lfs/git-lfs#3308]
When attempting to push various files to Artifactory via GitLFS, occasionally a file will not actually upload. The file will not be on Artifactory, but a pointer file will still be created for it.

The upload process appears as normal with nothing special outputted.

When attempting to pull the large file (we are using --skip-smudge most of the time and only pulling select files when needed), we get an error message like so: (the sha is fake, fyi)
```
git lfs pull -I path\to\the\file.7z
[5649684986g484986498498rg4s8498498498dg4f984984984984dfg6542jufs] Object does not exist: [404] Object does not exist
```
The file corresponding to that sha does not appear on Artifactory. When attempting to view the file on GitHub, it "helpfully" attempts to show the actual file contents: 
![image](https://user-images.githubusercontent.com/16566398/46689589-182e3280-cbb5-11e8-916c-5aba1dbda135.png)
Where a normal pointer file for a gitlfs file looks like this:
```
version https://git-lfs.github.com/spec/v1
oid sha256:5649684986g484986498498rg4s8498498498dg4f984984984984dfg6542jufs
size 116868493
```

The pointer file does exist in the repo, cloning it after a `git lfs install --skip-smudge` shows a regular pointer file. But the sha here is nowehere on Artifactory, and as far as I can tell, the file isn't uploaded at all. But I would expect an error message of some kind if the file were uploaded incorrectly or it failed to upload.

Is this an upload issue on my end, or is this a behavior of GitHub I can stop?

**Comment by bk2204** [comment]
Hey,

When you check a file into the repo and Git LFS is enabled, the pointer file is created, and the large object is written into the `.git` directory.  When an upload occurs, the `pre-push` hook ensures that all of your objects are uploaded and written to the remote system.  There's also a verify endpoint (that the server may or may not support, at its option) that will check that the file has been uploaded properly.

Usually the reason the LFS files in an upload don't get pushed is because the `pre-push` hook is missing; Git LFS tries very hard not to let a push happen if the LFS files are missing.  It's possible that someone pushed with `--no-verify`, which would bypass this, or that the hook was removed.  Usually running `git lfs install` in the repo will fix the situation if the hook got removed for some reason.

It's also possible that the upload failed at some point without an error and Artifactory doesn't implement the verify endpoint (I don't know whether it does), and so the check that the file should exist didn't occur.  This is very unlikely, but possible.

GitHub will, as of very recently, reject pushes without your LFS objects having been uploaded, so if you're using GitHub, the failure case will be much more obvious.

Can you check that the system that should have uploaded these changes has had `git lfs install` run in the repo and that the `pre-push` hook contains a call to `git lfs pre-push`?  That's the most likely culprit for this, and it's the easiest to fix, because you can have that system run `git lfs push --all` and it will likely upload that file for you.

**Comment by kirkpatt** [comment]
I believe when the file was uploaded, the `git lfs install` was run before the clone. Could that have caused the issue? Either way, reuploading the file from a repo after `git lfs install`-ing inside fixed the issue. Thank you for your help!

GitHub is still trying to show the contents, but the file is now valid. Any chance there's a way to keep GitHub from doing that?

**Comment by bk2204** [comment]
Yeah, it's possible that `git lfs install` never got the hooks set up, so the `pre-push` hook never got run.  I'm glad you got it sorted and found all your data.

I don't know the answer to your question getting GitHub to not show the contents; you'd probably be better off asking support@github.com, who will likely have a better answer than I will.

**Comment by pxwee5** [comment]
I am having this issue as well for some reason.
